ps3.py

```diff
@@ -0,0 +1,361 @@
+    """
+    Displays the letters currently in the hand.
+
+    For example:
+       display_hand({'a':1, 'x':2, 'l':3, 'e':1})
+    Should print out something like:
+       a x x l l l e
+    The order of the letters is unimportant.
+
+    hand: dictionary (string -> int)
+    """
+    print("Current Hand:", end=' ')
+    for letter in hand.keys():
+        for j in range(hand[letter]):
+            print(letter, end=' ')  # print all on the same line
+    print()  # print an empty line
+
+
+def deal_hand(n):
+    """
+    Returns a random hand containing n lowercase letters.
+    ceil(n/3) letters in the hand should be VOWELS (note,
+    ceil(n/3) means the smallest integer not less than n/3).
+
+    Hands are represented as dictionaries. The keys are
+    letters and the values are the number of times the
+    particular letter is repeated in that hand.
+
+    n: int >= 0
+    returns: dictionary (string -> int)
+    """
+
+    hand = {WILDCARD: 1}
+    num_vowels = int(math.ceil(n / 3))
+    for i in range(num_vowels - 1):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1
+    for i in range(num_vowels, n):
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+    return hand
+
+
+def update_hand(hand, word):
+    """
+    Does NOT assume that hand contains every letter in word at least as
+    many times as the letter appears in word. Letters in word that don't
+    appear in hand should be ignored. Letters that appear in word more times
+    than in hand should never result in a negative count; instead, set the
+    count in the returned hand to 0 (or remove the letter from the
+    dictionary, depending on how your code is structured).
+
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+
+    Has no side effects: does not modify hand.
+
+    word: string
+    hand: dictionary (string -> int)
+    returns: dictionary (string -> int)
+    """
+    word = word.lower()
+    hand_modified = hand.copy()
+    for letters in word:
+        if letters in hand_modified.keys():
+            hand_modified[letters] -= 1
+    list_keys = list(hand_modified.keys())
+    for keys in list_keys:
+        if hand_modified[keys] <= 0:
+            hand_modified.pop(keys)
+    return hand_modified
+
+
+def is_valid_word(word, hand, word_list):
+    """
+    Returns True if word is in the word_list and is entirely
+    composed of letters in the hand. Otherwise, returns False.
+    Does not mutate hand or word_list.
+
+    word: string
+    hand: dictionary (string -> int)
+    word_list: list of lowercase strings
+    returns: boolean
+    """
+    word = word.lower()
+    for letters in word:
+        if letters not in hand.keys() or word.count(letters) > hand[letters]:
+            return False
+
+    if WILDCARD in word:
+        list_split = word.split(WILDCARD)
+        pattern = f'{list_split[0]}[{VOWELS}]{list_split[1]}'
```

Comment:
'abc\*efg'.replace('\*', '[aeoui]') 
That would be more simple

Reply:
done